src/opm/input/eclipse/Schedule/Network/ExtNetwork.cpp

```diff
@@ -86,24 +86,65 @@ void ExtNetwork::add_branch(Branch branch)
     this->m_branches.push_back( std::move(branch) );
 }
 
-void ExtNetwork::drop_branch(const std::string& uptree_node, const std::string& downtree_node) {
-    auto downtree_branches = this->downtree_branches( downtree_node );
-    if (downtree_branches.empty()) {
+void ExtNetwork::add_or_replace_branch(Branch branch)
+{
+    const std::string& uptree_node = branch.uptree_node();
+    const std::string& downtree_node = branch.downtree_node();
+
+    // Add any missing nodes
+    for (const std::string& nodename : { downtree_node, uptree_node }) {
+        if (!this->has_node(nodename)) {
+            this->m_nodes.insert_or_assign(nodename, Node{nodename});
+        }
+        if (!this->has_indexed_node_name(nodename)) {
+            this->add_indexed_node_name(nodename);
+        }
+    }
+    
+    // Remove any other branch uptree from downtree_node (gathering tree structure required)
+    auto uptree_link = this->uptree_branch( downtree_node );
+    if (uptree_link.has_value()){
+        const auto& old_uptree_node = uptree_link.value().uptree_node();
+        this->drop_branch(old_uptree_node, downtree_node);
+    }
+    
+    // Update existing branch if it exists
+    auto downtree_branches = this->downtree_branches( uptree_node );
+    if (!downtree_branches.empty()) {
         auto branch_iter = std::find_if(this->m_branches.begin(), this->m_branches.end(), [&uptree_node, &downtree_node](const Branch& b) { return (b.uptree_node() == uptree_node && b.downtree_node() == downtree_node); });
-        if (branch_iter != this->m_branches.end())
-            this->m_branches.erase( branch_iter );
+        if (branch_iter != this->m_branches.end()) {
+            *branch_iter = branch;
+            return;
+        }
+    }
+    
+    this->m_branches.push_back( std::move(branch) );
+}
+
+bool ExtNetwork::is_disconnected(const std::string& node_name) const {
+    if (!this->has_node(node_name)) return false;
+
+    if (this->downtree_branches(node_name).empty()) {
+        for (const auto& branch : this->m_branches)
+            if (branch.downtree_node() == node_name) return false;
 
-        this->m_nodes.erase( downtree_node );
-    } else {
-        for (const auto& branch : downtree_branches)
-            this->drop_branch(branch.uptree_node(), branch.downtree_node());
+        return true;
+    }
+    return false;
+}
+
+void ExtNetwork::drop_branch(const std::string& uptree_node, const std::string& downtree_node) {
```

Comment:
With this way of `drop_branch`, it is possible that the `downtree_node` will be hanging there without being part of any branch. Should we worry about it?
Even the `uptree_node` can also be hanging there without being connected to any branch, then they are not meaningful to be part of the network. 

Maybe we should use the function `is_disconnected()`  to remove the disconnected nodes that does not belong to any branches?

However, I can see nodes that does not belong to the network can work as fix-pressure nodes.   So this comment is for discussion. Do you see the need to keep the potentially dangling nodes?

Reply:
Yes, for the rerouting feature, having the nodes hanging there is required for possible re-introduction into network at a later stage (keeping the same node properties).